src/cs/Ssh/IO/SshProtocol.cs

```diff
@@ -160,19 +163,25 @@ internal async Task<string> ReadProtocolVersionAsync(CancellationToken cancellat
 				if (line.StartsWith("SSH-", StringComparison.Ordinal))
 				{
 					this.metrics.AddMessageReceived(i + 1);
-					return line;
+					return (line, proxyIPAddress);
+				}
+				else if (line.StartsWith("PROXY ", StringComparison.Ordinal))
+				{
+					var split = line.Split(' ');
+					if (split.Length > 2 && IPAddress.TryParse(split[2], out var ipAddress))
```

Comment:
If parsing the PROXY line fails, it will be best to abort the connection (throw a `SshConnectionException`) for security.